### PR TITLE
[#17] project observability - common component and order module

### PR DESCRIPTION
**Pull request by NoahHsu**
## Description
- adopting spring boot 3 observability on the order module
- [introducing article](https://noahhsu.medium.com/spring-boot-3-observability-monitor-application-on-the-method-level-8057abec5926)
## Changes
- add dependency docker-compose file (for Grafana, Tempo, Loki, Prometheus), modify from this [repo](https://github.com/marcingrzejszczak/observability-boot-blog-post/blob/main/docker-compose.yml)
- add an aspect to log around the spring bean method
- make the self-config OpenFeign client bring traceId in requests
- add observability to order module
- use `ObserveAroundMethodHandler` to log around observed spring bean methods
- update README.md
## Screenshot(Optional)
- default dashboard
image
- Logs, Traces, Metrics
截圖 2023-02-26 上午1 09 59
- Spring Boot Observability
download and modify from [template](https://grafana.com/grafana/dashboards/17175-spring-boot-observability/)
image

### CODE REVIEW

**Review by woodyf — COMMENTED** [review]
What a huge commit! I should start reviewing after your sharing session, there's too many new "things".

File: observation/src/main/java/org/example/event/sourcing/order/poc/observation/config/LogAspectConfig.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.example.event.sourcing.order.poc.observation.config;
+
+import org.example.event.sourcing.order.poc.observation.aop.AbstractLogAspect;
+import org.example.event.sourcing.order.poc.observation.aop.DefaultLogAspect;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
```
Comment (woodyf) [inline observation/src/main/java/org/example/event/sourcing/order/poc/observation/config/LogAspectConfig.java:9]: hi

**Review by woodyf — APPROVED** [review]
👍

**Comment by NoahHsu** [comment]
related to #40

**Pull request merged**